docs/deprecation.md

```diff
@@ -0,0 +1,43 @@
+# Deprecation Policy
+
+In order for any software to be reliable enough for production use, it needs to have a stable API. This means that
+users would have a high confidence that the way they use Kubevirt would not become obsolete, or at least would give
+the proper time to prepare if it does.
+
+This document aims to specify a process for several aspects of deprecation revolving Kubevirt. This is intended to
+be a "live document" which updates and refines itself over time.
+
+## Process
+
+### Feature Gates
+#### Introduction
+A feature gate should be introduced for every new feature, capability, configuration or any other change that
+is not yet stable and can jeopardize the cluster.
+
+A feature gate serves as:
+* A way to warn the user about a possible danger that could be caused by an unstable logic.
+* A way for the users to explicitly state that they would like to take the risk and enable the feature.
+* A way for the user to explicitly confirm that some feature / HW is available on the cluster.
+
+Over time the new features would stabilize and graduate their API versions (i.e. from alpha, to beta to stable). When
+that happens, and therefore the risk of jeopardizing the user decreases significantly, the feature should be enabled
+by default and the feature gate should be deprecated (see section below).
+
+#### Deprecation
```

Comment:
I find this page super helpful from K8s - https://kubernetes.io/docs/reference/using-api/deprecation-guide/.  Can you add the requirement to advertise these deprecation somewhere on KubeVirt's web page?

Reply:
Hey @rthallisey!

> I find this page super helpful from K8s - https://kubernetes.io/docs/reference/using-api/deprecation-guide/.

Yes, indeed. This should also remain our main reference in the future.

> Can you add the requirement to advertise these deprecation somewhere on KubeVirt's web page?

Nice one!
I've added a requirement to notify kubevirt-dev mailing list when deprecating a feature gate.
Since feature gate deprecation wouldn't affect any user (only pop up a warning) I'm not sure we need to advertise it further than that, WDYT?

Reply:
Sure, that's ok with me.  At some point we'll need to document the deprecation process for features and that's where I'd expect the notification to be on the webpage.